### PR TITLE
Convert `SessionStateScopeEnumerator` to struct

### PR DESCRIPTION
**Pull request by xtqqczze**
Fix [HLQ006: GetEnumerator() and GetAsyncEnumerator() should return an instance of a value-typed enumerator](https://github.com/NetFabric/NetFabric.Hyperlinq.Analyzer/blob/master/docs/reference/HLQ006_GetEnumeratorReturnType.md)

### CODE REVIEW

File: src/System.Management.Automation/engine/SessionStateScopeEnumerator.cs [hunk 1]
```diff
@@ -94,7 +95,7 @@ System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator()
 
         public void Dispose()
         {
-            Reset();
+            // Nothing to dispose
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/SessionStateScopeEnumerator.cs:98]: Do we still need the method?

Comment (xtqqczze) [reply src/System.Management.Automation/engine/SessionStateScopeEnumerator.cs:98]: Yes, `IEnumerator<T>` implements `IDisposable`.

**Comment by vexx32** [comment]
Do we need to audit usages of the enumerator to ensure this won't cause issues in code areas that access the enumerator directly, if there are any?

**Comment by xtqqczze** [comment]
> Do we need to audit usages of the enumerator to ensure this won't cause issues in code areas that access the enumerator directly, if there are any?

How do you mean "access the enumerator directly"? Like this?

https://github.com/PowerShell/PowerShell/blob/658837323599ab1c7a81fe66fcd43f7420e4402b/src/System.Management.Automation/engine/SessionStateCmdletAPIs.cs#L54-L55

**Review by iSazonov — APPROVED** [review]
LGTM.

I suggest to add `#nullable enable` on top of SessionStateScopeEnumerator.cs file

**Comment by xtqqczze** [comment]
I'm not sure this change is worthwhile unless the enumerator can be refactored so that it is a ref struct, converting to draft.

**Comment by iSazonov** [comment]
> I'm not sure this change is worthwhile unless the enumerator can be refactored so that it is a ref struct, converting to draft.

I don't understand. Using struct for enumerator is best practice. Roslyng generates optimal code for this.

**Pull request closed without merging**